### PR TITLE
Tes 572 test steps

### PR DESCRIPTION
**Pull request by aamirtestable**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -347,8 +347,9 @@ execute(function(finished) {
 });
 
 if (fireNow) {
-	// trigger the event when smoke testing or run locally for testing
-	events.emit('my-event', 'MSFT');
+    // trigger the event when smoke testing or run locally for testing
+    events.emit('my-event', 'MSFT');
+    events.finish()
```
Comment (avistramer) [inline README.md:352]: This is only for smoke testing or running locally as indicated by the comment two lines above. Need the same thing in the actual example:

```
events.on('my-event', function(symbol) {
    request.get('http://sample.testable.io/stocks/' + symbol);
    events.finished();
    finished();
  });
```

File: README.md [hunk 2]
```diff
@@ -347,8 +347,9 @@ execute(function(finished) {
 });
```
Comment (avistramer) [inline README.md:348]: Need to add a new section in the README to document the suite/tests API you just added.

File: lib/events-watcher.js [hunk 3]
```diff
@@ -22,7 +22,8 @@ function watchForEvents(fileName, events, log) {
 	  	log.error('Error tailing live event', data);
 	});
 	tail.watch();
+  events.finish = () => { tail.unwatch(); }
```
Comment (avistramer) [inline lib/events-watcher.js:25]: Indentation here seems off from the rest of the file.

File: lib/main.js [hunk 4]
```diff
@@ -133,6 +135,22 @@ if (registerCommands) {
...
+  stepsObj.it = StepsClass.it.bind(StepsClass);
+  stepsObj.before = StepsClass.before.bind(StepsClass);
+  stepsObj.after = StepsClass.after.bind(StepsClass);
+  stepsObj.beforeEach = StepsClass.beforeEach.bind(StepsClass);
+  stepsObj.afterEach = StepsClass.afterEach.bind(StepsClass);
+
+  steps = stepsObj;
+}
+testSteps();
+
```
Comment (avistramer) [inline lib/main.js:153]: `StepsClass` is not a class it's an object. And in JS convention it should be camel cased. This also seems unnecessarily complicated.

```
const testSteps = new TestSteps();

module.exports.describe = testSteps.describe.bind(testSteps);
...
```

File: lib/testable-test.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+          flag: 'a'
+        }, resolve);
+      } else {
+        console.log('[Result] ' + JSON.stringify(result, null, '\t'));
+        resolve();
+      }
+    });
+  }
+
+  toErrorContent(error) {
```
Comment (avistramer) [inline lib/testable-test.js:233]: All these `if` statements are not applicable here. We know given the single location that calls this above that `error` will be an `Error` object. The first 5 if statements will never get called.

**Review by avistramer — CHANGES_REQUESTED** [review]
Need to update the typescript api (index.d.ts) and test this change from TS too.

File: lib/events-watcher.js (hunk 3 above)
Comment (aamirtestable) [reply lib/events-watcher.js:25]: Ah, tab vs spaces ... it was proper indented on idea IDE

**Comment by aamirtestable** [comment]
types updated

File: lib/testable-test.js [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+          else
+            resolve();
+        });
+      });
+    } else {
+      fn();
+    }
+  }
+
+  reportTestSuiteActions(type, contents) {
```
Comment (avistramer) [inline lib/testable-test.js:213]: For consistency with the other internal methods here this should be `_reportTestSuiteActions()` and `_toErrorTrace`.

**Review by avistramer — CHANGES_REQUESTED** [review]
One minor comment

**Review by avistramer — APPROVED** [review]

**Pull request merged**